### DISCUSSION TITLE
How to write the correct config file for GPT4All?

### DISCUSSION THREAD

**Discussion by AvengerMoJo** [explosion/spacy-llm#453] [Q&A] [answered]
[components.llm.model]
@llm_models = "langchain.GPT4All.v1"
name = "gemma-2b-it-q4_k_m.gguf"
query = {"@llm_queries": "spacy.CallLangChain.v1"}
config = {"temperature": 0.0}

As title, the above configuration doesn't work with Key model error

venv/lib/python3.12/site-packages/langchain_community/llms/gpt4all.py", line 141, in validate_environment
    full_path = values["model"]
                ~~~~~~^^^^^^^^^
KeyError: 'model'

Before I look into the code, I want to make sure I wrote the configuration file correctly. And I couldn't find the reference doc or example to use a local llm with GPT4All. 

Thanks

**Comment by AvengerMoJo** [comment]
I guess nobody has use GPT4All ?

**Reply by cediorio** [reply]
I've been trying to figure this out for Ollama and came across your post. No answer for you, but if I find one, I'll post back here...

**Comment by AvengerMoJo** [comment]
The answer is removing the 
config = {"temperature": 0.0}
I am still not sure why exactly but it fix the error.

**Comment by AvengerMoJo** [comment, marked as answer]
[nlp]
lang = "en"
pipeline = ["llm"]
batch_size = 1024


[components]

[components.llm]
factory = "llm"

[components.llm.task]
@llm_tasks = "spacy.TextCat.v3"
labels = ["COMPLIMENT", "INSULT"]

[components.llm.task.label_definitions]
"COMPLIMENT" = "a polite expression of praise or admiration.",
"INSULT" = "a disrespectful or scornfully abusive remark or act."
exclusive_classes = True
allow_none = False

#[components.llm.task.examples]
#@misc = "spacy.FewShotReader.v1"
#path = "example.json"

[components.llm.model]
@llm_models = "langchain.GPT4All.v1"
# name has to be folllow , can't use full path.  ~/.cache/gpt4all/
name = "gemma-2b-it-q4_k_m.gguf"
query = {"@llm_queries": "spacy.CallLangChain.v1"}
context_length = 250
